### PR TITLE
[new release] lbfgs (0.9.5)

### PR DESCRIPTION
**Pull request by Chris00**
Bound-constrainted optimization in many variables

- Project page: <a href="https://github.com/Chris00/L-BFGS-ocaml">https://github.com/Chris00/L-BFGS-ocaml</a>
- Documentation: <a href="https://Chris00.github.io/L-BFGS-ocaml/doc">https://Chris00.github.io/L-BFGS-ocaml/doc</a>

##### CHANGES:

- Compatibility with OCaml 5 (Edwin Török).
- Fix bug allowing to use it in the toplevel (Edwin Török).

### CODE REVIEW

File: packages/lbfgs/lbfgs.0.9.5/opam [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+authors: [ "Christophe Troestler <Christophe.Troestler@umons.ac.be>" ]
+license: "LGPL-3.0 with OCaml linking exception"
+homepage: "https://github.com/Chris00/L-BFGS-ocaml"
+dev-repo: "git+https://github.com/Chris00/L-BFGS-ocaml.git"
+bug-reports: "https://github.com/Chris00/L-BFGS-ocaml/issues"
+doc: "https://Chris00.github.io/L-BFGS-ocaml/doc"
+tags: [ "science" "numerics" "optimization" "minimization" "maximization"
+        "bound-constrained" "large-scale" ]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (haochenx) [inline packages/lbfgs/lbfgs.0.9.5/opam:12]:
```suggestion
  ["dune" "subst"] {dev}
```

This causes a linter warning. I believe if you use Dune >= 2.7 to make the publish, it should automatically use `dev` instead of `pin` for this clause.
ref: https://github.com/ocaml/dune/blob/main/CHANGES.md#270-13082020

File: packages/lbfgs/lbfgs.0.9.5/opam [hunk 2]
```diff
@@ -0,0 +1,39 @@
+opam-version: "2.0"
+maintainer: "Christophe Troestler <Christophe.Troestler@umons.ac.be>"
+authors: [ "Christophe Troestler <Christophe.Troestler@umons.ac.be>" ]
+license: "LGPL-3.0 with OCaml linking exception"
```
Comment (haochenx) [inline packages/lbfgs/lbfgs.0.9.5/opam:4]: it seems that the previous license is `LGPL-3.0-only WITH OCaml-LGPL-linking-exception` (which is slightly different than the one in this PR.) Is this change intentional?

**Comment by haochenx** [comment]
The CI failures seem unrelated to this package and are not blockers.

**Comment by haochenx** [comment]
just fyi: @mmottl 
it seems that lacamel is failing to build on ppc64-ocaml-4.14 and you may want to have a look
image
```
The following actions will be performed:
=== install 2 packages
  - install lacaml 11.0.10        [required by lbfgs]
  - install lbfgs  0.9.5 (pinned)

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
Processing  1/6:
-> retrieved lacaml.11.0.10  (cached)
Processing  3/6: [lacaml: dune build]
+ /home/opam/.opam/opam-init/hooks/sandbox.sh "build" "dune" "build" "-p" "lacaml" "-j" "175" "@install" (CWD=/home/opam/.opam/4.14/.opam-switch/build/lacaml.11.0.10)
- File "src/config/dune", line 24, characters 35-57:
- 24 |     (names zdot_is_procedure_stubs zdot_is_function_stubs)
-                                         ^^^^^^^^^^^^^^^^^^^^^^
- (cd _build/default/src/config && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -D_FILE_OFFSET_BITS=64 -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -DEXTERNAL_EXP10 -std=c99 -march=native -O3 -ffast-math -fPIC -DPIC -g -I /home/opam/.opam/4.14/lib/ocaml -o zdot_is_function_stubs.o -c zdot_is_function_stubs.c)
- gcc: error: unrecognized command-line option '-march=native'; did you mean '-mcpu=native'?
- File "src/config/dune", line 24, characters 11-34:
- 24 |     (names zdot_is_procedure_stubs zdot_is_function_stubs)
-                 ^^^^^^^^^^^^^^^^^^^^^^^
- (cd _build/default/src/config && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -D_FILE_OFFSET_BITS=64 -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -DEXTERNAL_EXP10 -std=c99 -march=native -O3 -ffast-math -fPIC -DPIC -g -I /home/opam/.opam/4.14/lib/ocaml -o zdot_is_procedure_stubs.o -c zdot_is_procedure_stubs.c)
- gcc: error: unrecognized command-line option '-march=native'; did you mean '-mcpu=native'?
```

File: packages/lbfgs/lbfgs.0.9.5/opam (hunk 1 above)
Comment (Chris00) [reply packages/lbfgs/lbfgs.0.9.5/opam:12]: Thanks.

**Review by haochenx — APPROVED** [review]
Thanks! Looks good to me now. CI failures are all unrelated.
